packages/flutter/lib/src/material/ink_ripple.dart

```diff
@@ -245,7 +249,7 @@ class InkRipple extends InteractiveInkFeature {
     if (_clipCallback != null) {
       final Rect rect = _clipCallback();
       if (_customBorder != null) {
-        canvas.clipPath(_customBorder.getOuterPath(rect));
+        canvas.clipPath(_customBorder.getOuterPath(rect, textDirection: _textDirection));
```

Comment:
Is it understood why a missing `@required` parameter wasn't caught by the various build checks & analysis? For that matter why is `textDirection` a named parameter in this case?

Reply:
The textDirection isn't `@required` in ShapeBorder, and that is the static type of `_customBorder` here.